Add Memgraph integration

matea16 opened the pull request:
# Description

Add Memgraph integration. Extended from the **`KnowledgeGraphIndex`** and **`PropertyGraphIndex`** class.

## New Package?

Did I fill in the `tool.llamahub` section in the `pyproject.toml` and provide a detailed README.md for my new integration or package?

- [x] Yes
- [ ] No

## Version Bump?

Did I bump the version in the `pyproject.toml` file of the package I am updating? (Except for the `llama-index-core` package)

- [ ] Yes
- [x] No

## Type of Change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [x] Added new unit/integration tests
- [ ] Added new notebook (that tests end-to-end)
- [ ] I stared at the code and made sure it makes sense

## Suggested Checklist:

- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have added Google Colab support for the newly added notebooks.
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [ ] I ran `make format; make lint` to appease the lint gods

logan-markewich commented on llama-index-integrations/graph_stores/llama-index-graph-stores-memgraph/README.md:
Lets put the example installation and basic example usage/setup here -- this shows up directly on pypi and llamahub.ai for the package

logan-markewich commented on llama-index-integrations/graph_stores/llama-index-graph-stores-memgraph/llama_index/graph_stores/memgraph/property_graph.py:
```diff
@@ -0,0 +1,927 @@
+        refresh_schema: bool = True,
+        sanitize_query_output: bool = True,
+        enhanced_schema: bool = False,
+        **neo4j_kwargs: Any,
```
nit: probably don't want to mention neo4j here 😉

logan-markewich replied:
oh, its using the neo4j client --interesting, ok!

logan-markewich commented on llama-index-integrations/graph_stores/llama-index-graph-stores-memgraph/pyproject.toml:
```diff
@@ -0,0 +1,57 @@
+
+[tool.poetry.dependencies]
+python = ">=3.8.1,<4.0"
+llama-index-core = "^0.10.0"
```
neo4j should be a dependency I think

logan-markewich commented:
For unit tests, you'll either need to mock the neo4j client, or use pytest.skipif to skip tests if the client can't connect to neo4j

matea16 commented:
Hi @logan-markewich, thank you for the quick response and feedback. I've implemented all of your suggestions, moved a basic usage example to the README file, added a more detailed explanation to the jupyter notebook example for the docs and improved the tests. Let me know if this works, looking forward to the integration :)
